mne/decoding/time_gen.py

```diff
@@ -203,16 +224,31 @@ def predict(self, epochs):
         if not hasattr(self, 'estimators_'):
             raise RuntimeError('Please fit models before trying to predict')
 
-        # clean attributes
+        # Check predict mode
+        if self.predict_mode not in ['cross-validation', 'mean-prediction']:
+            raise ValueError('`predict_mode` must be a str, "mean-prediction" '
+                             'or "cross-validation"')
+
+        # Check that training cv and predicting cv match
+        if self.predict_mode == 'cross-validation':
+            n_est_cv = [len(estimator) for estimator in self.estimators_]
+            inconsistent_splits = len(set(n_est_cv)) != 1
+            mismatching_splits = n_est_cv[0] != len(self._cv_splits)
+            mismatching_y = len(self.y_train_) != len(epochs)
+            if inconsistent_splits or mismatching_splits or mismatching_y:
+                raise ValueError(
+                    'When `predict_mode = "cross-validation"`, the training '
+                    'and predicting cv schemes must be identical.')
+
+        # Clean attributes
         for att in ['y_pred_', 'test_times_', 'scores_', 'scorer_', 'y_true_']:
             if hasattr(self, att):
                 delattr(self, att)
-
-        n_jobs = self.n_jobs
+        _warn_once.clear()  # reset self-baked warning tracker
 
         X, y, _ = _check_epochs_input(epochs, None, self.picks_)
 
-        if not np.all([len(test) for train, test in self.cv_]):
```

Comment:
`cv_` is still there. We use `_cv_splits` to collect train and test in cv once, and avoid non deterministic, and backward incompatibilities issues.